types/mailchimp__mailchimp_marketing/index.d.ts

```diff
@@ -31,189 +71,383 @@ export interface Body {
+        gmtoff: number;
+        dstoff: number;
+        country_code: string;
+        timezone: string;
+        region: string;
+    }
+
+    interface MemberMarketingPermissions extends MemberMarketingPermissionsInput {
+        text: string;
+    }
+
+    interface MemberMarketingPermissionsInput {
+        marketing_permission_id: string;
+        enabled: boolean;
+    }
+
+    interface MemberLastNote {
+        note_id: number;
+        created_at: string;
+        created_by: string;
+        note: string;
+    }
+
+    interface Tags {
+        id: number;
+        name: string;
+    }
+
+    /**
+     * Add or update a list member.
+     * @param listId The unique ID for the list.
+     * @param subscriberHash The MD5 hash of the lowercase version of the list member's email address. This endpoint also accepts a list member's email address or contact_id.
+     * @param body
+     * @param opts Optional parameters
+     * @param opts.skipMergeValidation If skip_merge_validation is true, member data will be accepted without merge field values, even if the merge field is usually required. This defaults to false.
+     * @return A {@link https://www.promisejs.org/|Promise}, with data of type {@link module:model/ListMembers2}
+     */
+    function setListMember(listId: string, subscriberHash: string, body: SetListMemberBody, opts?: ListOptions): Promise<MembersSuccessResponse | ErrorResponse>;
+
+    /**
+     * Get information about a specific list member, including a currently subscribed, unsubscribed, or bounced member.
+     * @param listId The unique ID for the list.
+     * @param subscriberHash The MD5 hash of the lowercase version of the list member's email address. This endpoint also accepts a list member's email address or contact_id.
+     * @param opts Optional parameters
+     * @param opts.fields A comma-separated list of fields to return. Reference parameters of sub-objects with dot notation.
+     * @param opts.excludeFields A comma-separated list of fields to exclude. Reference parameters of sub-objects with dot notation.
+     * @return A {@link https://www.promisejs.org/|Promise}, with data of type {@link module:model/ListMembers2}
+     */
+    function getListMember(listId: string, subscriberHash: string, opts?: ListOptions): Promise<MembersSuccessResponse | ErrorResponse>;
+
+    /**
+     * Add a new member to the list.
+     * @param listId The unique ID for the list.
+     * @param body
+     * @param opts Optional parameters
+     * @param opts.skipMergeValidation If skip_merge_validation is true, member data will be accepted without merge field values, even if the merge field is usually required. This defaults to false.
+     * @return A {@link https://www.promisejs.org/|Promise}, with data of type {@link module:model/ListMembers2}
+     */
+    function addListMember(listId: string, body: AddListMemberBody, opts?: ListOptions): Promise<MembersSuccessResponse | ErrorResponse>;
```

Comment:
Hi,

I'm using these types in a project however am facing issues with the `ErrorResponse` type. In the [MailChimp package](https://github.com/mailchimp/mailchimp-marketing-node/blob/master/src/ApiClient.js#L386), if the api request returns an error code (`400 - 499`) the Promise will `reject` which cannot be typed. Should the `ErrorResponse` type be on these methods since a `catch` error cannot be typed either?

Thanks,
Fraser

Reply:
@rattkin @jeremybarbet @odanieldcs @edwinfinch

He is right on this, should we update the type signatures?